lambdaclass/afk_gacha_game
# [GH-437] Claim dungeon AFK Rewards

Closes #437 

## Motivation

We need to claim the AFK rewards every time the user goes to the Dungeon Settlement

## Summary of changes

- Claim AFK rewards when entering the Dungeon Settlement scene

## How has this been tested?

1. Start the backend using the main branch. Before doing it, you may want to edit `apps/champions/priv/dungeon_settlement_levels.json` and increase the amount of afk supplies provided by the first Dungeon level. Changing it to 1440 will give you 1 supply per minute, and you won't need to wait that long to test this behaviour.
2. Get your user in the backend and check the initial amount of supplies. To do so, you can run `{:ok, user} = Champions.Users.get_user(user_id)` and then `user.currencies` to see the available currencies.
3. Wait a few moments (depending of what you did in step 1) and go to the Dungeon Settlement scene.
4. Get your user from the backend again, and check that the claimed supplies were added.

You'll notice that if you come back and forth from that scene during the waiting time, you won't receive any rewards despite how long you wait. We'll address this problem in this issue: https://github.com/lambdaclass/afk_gacha_game/issues/491

## Test additions / changes

List tests added/updated.

## Checklist
